rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/FileMetadata.java

```diff
@@ -49,7 +50,7 @@
   private final long lastModified;
   private long currentFileSize;
   private boolean needsRefresh = true;
-  private final int generationNumber;
+  private final ImmutablePair<Integer, Boolean> generationNumber;
```

Comment:
Continue storing this as `int`, we dont need information about reuse outside constructor.